lib/context.ml

```diff
@@ -69,7 +68,7 @@ let is_pipeline_allowed ctx repo_url ~pipeline =
   | None -> true
   | Some config ->
   match config.status_rules.allowed_pipelines with
-  | Some allowed_pipelines when not @@ List.exists allowed_pipelines ~f:(String.equal pipeline) -> false
+  | Some allowed_pipelines when not @@ List.mem pipeline allowed_pipelines -> false
```

Comment:
`List.mem` uses `(=)` to test equality, consider using `List.find_opt` or `List.exists` to be more explicit on which equal function to use.